W-14137568-set-payload-cv (#176)
mulesoft/docs-code-builder: modules/ROOT/pages/_partials/mule-components/4.4/component-set-payload.adoc

//
//tag::component-title[]

= Set Payload (<set-payload/>)

//end::component-title[]
//

//
//tag::component-short-description[]
//     Short description of the form "Do something..." 
//     Example: "Configure log messages anywhere in a flow."

> Updates the payload of the message.

Reviewer: Is it always updating? Maybe "Configures the payload of" ?

> Updates the payload of the message.

Reviewer: ```suggestion
Sets or updates the payload of the message.
```

//end::component-short-description[]
//

//
//tag::component-long-description[]

> The payload can be a literal string or a DataWeave expression. Use the `set-payload` component for simple expressions or transformation. 

Reviewer: ```suggestion
The payload can be a literal string or a DataWeave expression. Use the `set-payload` component only for simple expressions or transformations.
```

> //For complex scenarios, use the xref:acb-component-transform.adoc[Transform Message Component].

Reviewer: ```suggestion
For complex scenarios, use the xref:acb-component-transform.adoc[Transform Message Component].
```

//end::component-long-description[]
//

//SECTION: COMPONENT XML
//
//tag::component-xml-title[]

[[component-xml]]
== Component XML

This component supports the following XML structure:

//end::component-xml-title[]
//
//
//tag::component-xml[]

[source,xml]
----
<set-payload 
  value="" 
  doc:name="Set payload" 
  doc:id="vnjpqo" />
----

//end::component-xml[]
//
//tag::component-snippet-xml[]

[[snippet]]

=== Snippet XML

Snippet XML in Anypoint Code Builder serves as a customizable template for the component. 

[source,xml]
----
SNIPPET_XML_HERE
----

//end::component-snippet-xml[]
//
//
//
//
//TABLE: ROOT XML ATTRIBUTES (for the top-level (root) element)
//tag::component-xml-attributes-root[]

[%header,cols="1,3a"]
|===
| `<set-payload/>` Attributes 
| Description

| `doc:name` 
| Editable name for the component.

| `doc:id` 
| Automatically generated identifier for the component.

> | `value` 
> | Accepts a literal string or DataWeave expression that defines how to set the payload. For example, ``"some string"`` or `#[now()]`. This attribute is required. 

Author: ```suggestion
| `value` 
| Accepts a literal string or DataWeave expression that defines how to set the payload. For example, ``"some string"`` or `#[now()]`. This attribute is required. 

| `resource`
|  Points to a `.dwl` file containing a DataWeave transformation script. These values are resolved statically (not dynamically). For example, using `<ee:set-payload resource="myscript-${env}.dwl"/>` generates an error.
```

| `mimeType` 

> | The mime type of the value assigned to the payload. For example, `text/plain` or `application/json`.

Reviewer: ```suggestion
| The MIME type of the value assigned to the payload. For example, `text/plain` or `application/json`.
```

| `encoding`
| The encoding of the value assigned to the payload. For example, `UTF-8`.

|===

[WARNING]
--

> The `mimeType` and `encoding` attributes do not affect a DataWeave expression used as `value`. They only affect the output Mule Message. If a transformation is required, the DataWeave expression must contain an explicit output directive.

Reviewer: ```suggestion
The `mimeType` and `encoding` attributes do not affect a DataWeave expression used as a `value`. These attributes affect the _output_ Mule Message only. If a transformation is required, the DataWeave expression must contain an explicit `output` directive.
```
